nncf/tensorflow/quantization/quantizers.py

```diff
@@ -149,17 +150,16 @@ def setup_input_transformation(self, input_shape, input_type, input_name, layer)
         the original inputs shape.
 
         :param input_shape: Shape of the input.
-        :param input_type: Type of the input identifies that inputs are layer weights
-            or inputs of the layer.
-        :param input_name: Input name.
-        :param layer: Layer, where the Quantizer is registered.
+        :param channel_axes: Channel axes.
         """
-        self._pre_processing_fn, self._post_processing_fn = \
-            self._make_transformation_fns(input_shape, input_type, input_name, layer)
-
-    def _make_transformation_fns(self, input_shape, input_type, input_name, layer):
-        channel_axes = get_channel_axis(input_type, input_name, layer)
+        try:
+            self._pre_processing_fn, self._post_processing_fn = \
+                Quantizer._make_transformation_fns(input_shape, channel_axes)
+        except NotImplementedError as e:
+            raise NotImplementedError(f'Additional information: quantizer name {self.name}') from e
 
+    @staticmethod
```

Comment:
Could you briefly explain why you made the `_make_transformation_fns` function `staticmethod`?

Reply:
The `_make_transformation_fns` method doesn't use the internal state of the instance (`self` parameter). The `self` parameter was only needed to call the `_make_pre_processing_fn` and `_make_post_processing_fn` methods. But these methods are static and we can use the class name to call it. I think this change makes the code more readable. We immediately see that the `_make_transformation_fns` doesn't use/change the state of the instance. This is an auxiliary code that could be taken out of the class. But for historical reasons, we write all code inside the class.

If you have any concerns about this, I will revert these changes.

Reply:
I can guess that there may be problems if we want to override, for example the `_make_transformation_fns` function in the descendant class. This is refactoring and I don't see need to change it in this PR.

Reply:
I don't see a reason to change something.